activesupport/test/time_travel_test.rb

```diff
@@ -77,6 +77,49 @@ def test_time_helper_travel_back
     end
   end
 
+  def test_time_helper_travel_to_with_nested_calls_with_blocks
+    Time.stub(:now, Time.now) do
+      outer_expected_time = Time.new(2004, 11, 24, 01, 04, 44)
+      inner_expected_time = Time.new(2004, 10, 24, 01, 04, 44)
+      travel_to outer_expected_time do
+        error = assert_raises(RuntimeError) do
+          travel_to(inner_expected_time) do
+            #noop
+          end
+        end
+        assert_equal "\n      Calling `travel_to` with a block, when we have previously already made a call to `travel_to`, can lead to confusing time stubbing.\n\n      Instead of:\n\n         travel_to 2.days.from_now do\n           # 2 days from today\n           travel_to 3.days.from_now do\n             # 5 days from today\n           end\n         end\n\n      preferred way to achieve above is:\n\n         travel_to 2.days.from_now do\n           # 2 days from today\n         end\n\n         travel_to 5.days.from_now do\n           # 5 days from today\n         end\n\n", error.message
+      end
+    end
+  end
+
+  def test_time_helper_travel_to_with_nested_calls
+    Time.stub(:now, Time.now) do
+      outer_expected_time = Time.new(2004, 11, 24, 01, 04, 44)
+      inner_expected_time = Time.new(2004, 10, 24, 01, 04, 44)
+      travel_to outer_expected_time do
+        assert_nothing_raised do
+          travel_to(inner_expected_time)
+          assert_equal inner_expected_time, Time.now
+        end
+      end
+    end
+  end
+
+  def test_time_helper_travel_to_with_subsequent_calls
+    Time.stub(:now, Time.now) do
+      initial_expected_time = Time.new(2004, 11, 24, 01, 04, 44)
+      subsequent_expected_time = Time.new(2004, 10, 24, 01, 04, 44)
+      assert_nothing_raised do
+        travel_to initial_expected_time
+        travel_to subsequent_expected_time
+        travel_to initial_expected_time
+        travel_to subsequent_expected_time
```

Comment:
Do we need to do the initial subsequent dance a second time? Could do with some newlines :)